Projects/Shapes/Shapes/Shapes/DynamicArray.h

```diff
@@ -0,0 +1,339 @@
+#pragma once
+
+template<typename T>
+class TDynamicArray
+{
+private:
+	int Capacity = 0;
+	int Size = 0;
+	T* Array;
+
+public:
+
+	//-------------------------------------------------------------------------------------------
+	//								CONSTRUCTORS & DESTRUCTORS
+	//-------------------------------------------------------------------------------------------
+	
+	TDynamicArray()
+	{
+		
+	}
+
+	TDynamicArray(const int InCapacity)
+	{
+		Capacity = InCapacity;
```

Comment:
Capacity and Size are by default 0, initialized in the member variables above, according to our coding standards. The two constructors available for TDynamicArray are:
- Default Constructor -> This one is empty, as it would be a waste if it initialized anything by default. If I were to make an Array of N TDynamicArrays, it would instantly call N constructors. If the default constructor reserves something by default, the time and space complexity would be affected.
- Capacity Constructor -> This one can be called if you already have a good estimate of the capacity your array is going to need. This allows for optimization in space complexity. If you are unsure of the capacity it should initially have, it would be better to call the Default Constructor.

The downside to this structure of constructors is the additional if clause check in the Push function. However, the time complexity loss there is negligible compared to that saved by the default constructor change.